src/rest_vol_datatype.c

```diff
@@ -328,7 +328,7 @@ RV_datatype_open(void *obj, const H5VL_loc_params_t *loc_params, const char *nam
 {
     RV_object_t *parent   = (RV_object_t *)obj;
     RV_object_t *datatype = NULL;
-    H5I_type_t   obj_type = H5I_UNINIT;
```

Comment:
Other object open callbacks appear to still be using H5I_UNINIT. Does this work correctly here? Does this need to be changed back or can the other object open callbacks set their expected types ahead of time now?

Reply:
`RV_find_object_by_path2` stores the type of the object it finds, but no longer needs the initial value when making the request.

`RV_find_object_by_path1` should still need H5I_UNINIT in some cases - it looks like `RV_datatype_open` isn't used by any of the active tests right now. I'll set this back to H5I_UNINIT for consistency with any version of HSDS.